### PR TITLE
change ready status to "online/offline"

### PR DESCRIPTION
**Pull request by jubrad**
### Motivation

        moves ready/not-ready to online/offline
        in order to avoid conflation with a future
        hydration status ready status

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Review by benesch — APPROVED** [review]
LGTM assuming SQL council signoff!

File: src/catalog/src/builtin.rs [hunk 1]
```diff
@@ -6198,17 +6198,18 @@ pub static MZ_SHOW_CLUSTER_REPLICAS: Lazy<BuiltinView> = Lazy::new(|| BuiltinVie
...
     mz_catalog.mz_cluster_replicas.name AS replica,
     mz_catalog.mz_cluster_replicas.size AS size,
-    statuses.ready AS ready
+    COALESCE(statuses.ready, false) AS ready 
 FROM
     mz_catalog.mz_cluster_replicas
         JOIN mz_catalog.mz_clusters
             ON mz_catalog.mz_cluster_replicas.cluster_id = mz_catalog.mz_clusters.id
-        JOIN
+        LEFT JOIN
```
Comment (jubrad) [inline src/catalog/src/builtin.rs:6206]: `mz_hydration_statuses` stores hydration statuses for installed dataflows, in the replica is not yet online it won't have installed dataflows so we need to left join here and do the coalesce above to coarse into a bool. "ready"=true now means online + hydrated, and "ready"=false means offline or not hydrated.

File: src/catalog/src/builtin.rs [hunk 2]
```diff
@@ -6198,17 +6198,18 @@ pub static MZ_SHOW_CLUSTER_REPLICAS: Lazy<BuiltinView> = Lazy::new(|| BuiltinVie
...
             (
                 SELECT
                     replica_id,
-                    mz_unsafe.mz_all(status = 'ready') AS ready
-                FROM mz_internal.mz_cluster_replica_statuses
+                    min(hydrated) AS ready
```
Comment (benesch) [inline src/catalog/src/builtin.rs:6210]: I find `bool_and` to be a bit clearer in situations like this!

```suggestion
                    bool_and(hydrated) AS ready
```

File: src/catalog/src/builtin.rs [hunk 3]
```diff
@@ -6198,17 +6198,18 @@ pub static MZ_SHOW_CLUSTER_REPLICAS: Lazy<BuiltinView> = Lazy::new(|| BuiltinVie
     mz_catalog.mz_clusters.name AS cluster,
     mz_catalog.mz_cluster_replicas.name AS replica,
     mz_catalog.mz_cluster_replicas.size AS size,
-    statuses.ready AS ready
+    COALESCE(statuses.ready, false) AS ready
```
Comment (benesch) [inline src/catalog/src/builtin.rs:6201]:
```suggestion
    coalesce(statuses.ready, FALSE) AS ready
```

(per STYLE.md)

**Review by benesch — APPROVED** [review]
LGTM, but please also update mz_internal.md in the docs to reflect the new states.

**Pull request merged**